packages/events/EventPluginHub.js

```diff
@@ -198,12 +209,13 @@ export function enqueueEvents(events) {
  *
  * @internal
  */
-export function processEventQueue(simulated) {
+export function processEventQueue(simulated: boolean) {
   // Set `eventQueue` to null before processing it so that we can tell if more
   // events get enqueued while processing.
   var processingEventQueue = eventQueue;
   eventQueue = null;
-  if (simulated) {
+
+  if (simulated && processingEventQueue) {
```

Comment:
This seems a bit odd. If we don't want to do anything when `processingEventQueue` is empty, can we exit early instead? Otherwise we're directing `simulated` events into non-simulated branch (even though technically there's no events) which seems wrong.